### PR TITLE
Enabled battery monitoring mode to get actual battery data

### PR DESCRIPTION
**Pull request by Elf2707**
## Description

Fixed issue #396

Enabled battery monitoring mode to get actual battery data

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [x] I added the documentation in `README.md`.
* [ ] I mentionned this change in `CHANGELOG.md`.

### CODE REVIEW

**Comment by Elf2707** [comment]
I've been thinking. Maybe a better way not to enable batteryMonitoringMode in each getBatteryLevel call like I did in this PR, but just leave this method as it is right now and just add a hint to documentation to put this line to AppDelegate.m application:didFinishLaunchingWithOptions to get battery level:

[UIDevice currentDevice].batteryMonitoringEnabled = true;

**Comment by machour** [comment]
@Elf2707 Thank you so much.
@anwarhamr could you take a look at this PR, and Elf2707's last comment and give us some feedback?

**Comment by anwarhamr** [comment]
Great that did the trick for and iPhone7.  Could this be something that is handled by the `react-native link` command?
On a side note, how do I see or know when the next release will be available?

**Comment by machour** [comment]
I'm not sure if `link` can handle this, but we would definitely add this to the README. 

@elf2707, as soon as you update the PR accordingly, I'll publish a new release. Thank you again!

**Comment by machour** [comment]
@anwarhamr, you'll definitely get a notification when github when I merge this one, and usually I notify all participants when I publish something they helped fixing. Thank you again for helping out!

**Comment by Elf2707** [comment]
Hi everyone. I changed getBatteryLevel back and added note to getBatteryLevel description. Its a right way to put [UIDevice currentDevice].batteryMonitoringEnabled = true; to AppDelegate. Because it should be called once per application not every getBatteryLevel call. Thanks.

File: ios/RNDeviceInfo.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (machour) [inline ios/RNDeviceInfo.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Is this file needed?

Comment (Elf2707) [reply ios/RNDeviceInfo.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Actually according to https://developer.apple.com/library/content/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html

>Xcode 9.3 adds a new IDEWorkspaceChecks.plist file to a workspace's shared data, to store >the state of necessary workspace checks. Committing this file to source control will prevent >unnecessary rerunning of those checks for each user opening the workspace. (37293167)

So I could delete it, but as I understand it is useful file. Should I delete it anyway?

**Comment by hickeyadamc** [comment]
I'm also having the issue where getBatteryLevel returns -1. Was this change ever merged in?

File: ios/RNDeviceInfo.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 1 above)
Comment (mikehardy) [reply ios/RNDeviceInfo.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Looks harmless and I confirm other high-quality modules have it

**Comment by mikehardy** [comment]
@machour I think this is good to merge. A valid documentation update and the other included change appears fine. You can put a "postlink" (and "postunlink") script in package.json to have link handle this, but altering the internals of a method in AppDelegate seems hard and potentially brittle. I think the documentation pointer is sufficient

**Comment by machour** [comment]
react-native-device-info 0.27.1 published 🎉

**Comment by mikehardy** [comment]
> react-native-device-info 0.27.1 published 🎉

Boom! cleared a lot out in one go, awesome. I'll keep an ear out for issues
